src/ansys/geometry/core/tools/problem_areas.py

```diff
@@ -109,11 +112,14 @@ def fix(self) -> RepairToolMessage:
         response = self._repair_stub.FixDuplicateFaces(
             FixDuplicateFacesRequest(duplicate_face_problem_area_id=self._id_grpc)
         )
+        parentdesign = self.faces[0].body.parent_component
```

Comment:
@umutsoysalansys this is the way to go but it's not enough yet, since there can be a hierarchy of component, you should iterate on parent_component's parent_component until you get "None" as parent_component's value. When you do, that's your design. If this explanation seems confusing, feel free to reach out, we'll se that together.